clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/view/EcsApplicationProvider.java

```diff
@@ -0,0 +1,147 @@
+package com.netflix.spinnaker.clouddriver.ecs.view;
+
+
+import com.amazonaws.services.ecs.AmazonECS;
+import com.amazonaws.services.ecs.model.*;
+import com.google.common.collect.Sets;
+import com.netflix.spinnaker.clouddriver.aws.security.AmazonClientProvider;
+import com.netflix.spinnaker.clouddriver.aws.security.AmazonCredentials;
+import com.netflix.spinnaker.clouddriver.ecs.model.EcsApplication;
+import com.netflix.spinnaker.clouddriver.model.Application;
+import com.netflix.spinnaker.clouddriver.model.ApplicationProvider;
+import com.netflix.spinnaker.clouddriver.security.AccountCredentials;
+import com.netflix.spinnaker.clouddriver.security.AccountCredentialsProvider;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+@Component
+public class EcsApplicationProvider implements ApplicationProvider {
+
+  public static final String SERVICE_DELIMITER = "-";
+
+  AccountCredentialsProvider accountCredentialsProvider;
+  AmazonClientProvider amazonClientProvider;
+
+  @Autowired
+  public EcsApplicationProvider(AccountCredentialsProvider accountCredentialsProvider, AmazonClientProvider amazonClientProvider) {
+    this.accountCredentialsProvider = accountCredentialsProvider;
+    this.amazonClientProvider = amazonClientProvider;
+    getApplications(false);
+  }
+
+
+  @Override
+  public Application getApplication(String name) {
+
+    for (Application application: getApplications(false)) {
+      if (name.equals(application.getName())) {
```

Comment:
NPE on name=null, is application.getName() guaranteed to have a non-null String?

Reply:
name would not be null if called from the GUI
application.getName() is probably non-null, but that probability is lower than when that is called from the GUI